src/components/Modals/ClearSemester.vue

```diff
@@ -24,14 +24,9 @@ import eraseIcon from '@/assets/images/erase-white.svg';
 
 export default defineComponent({
   components: { TeleportModal },
-  props: {
-    clearSemType: { type: String, required: true },
-    clearSemYear: { type: Number, required: true },
-  },
   emits: {
     'close-clear-sem': () => true,
-    'clear-semester': (type: string, year: number) =>
-      typeof type === 'string' && typeof year === 'number',
+    'clear-semester': () => true,
```

Comment:
These are redundant props because the `<clear-semester>` component is a child of the `<semester>` prop.

I think it was modeled after `delete-semester`, which does need a season & year passed in to delete the semester (because it is emitted to the grandparent `<semester-view>`). Instead, `clear-semester` is more similar to `add-course` and `delete-course`, despite what the naming suggests.